src/common/utils/io.ts

```
@@ -17,3 +17,30 @@ export const resolveS3Path = (path: string, fileclass: string) => {
       throw new Error('fileclass must be one of "texture", "model".');
   }
 };
+
+/**
+ * serialize is a helper function that is a light wrapper around stringifying a JSON object.
+ * @param data
+ * @returns string
+ */
+export function serialize(data: Object): string | undefined {
+  try {
+    return JSON.stringify(data);
+  } catch (e) {
+    return undefined;
```

Can we console.warn or something here so it's not a silent failure?

---

good catch, nice work catching this issue from old code